pkgs/development/compilers/dmd/default.nix

```diff
@@ -47,9 +126,11 @@ stdenv.mkDerivation rec {
 
       cd ../phobos
       mkdir $out/lib
-      ${let bits = if stdenv.is64bit then "64" else "32";
-            osname = if stdenv.isDarwin then "osx" else "linux"; in
-      "cp generated/${osname}/release/${bits}/libphobos2.a $out/lib"
+      ${
+          let bits = builtins.toString stdenv.hostPlatform.parsed.cpu.bits;
+          osname = if stdenv.isDarwin then "osx" else "linux";
```

Comment:
Your second there is correct---if the expression evaluates to the same value, there is no rebuild. `stdenv.hostPlatform.parsed.kernel.name` evaluates to `"linux"` on Linux, so there is no change!

Thanks so much for these fixes, but don't forgot the `stdenv.hostPlatform.is*` change. `stdenv.is*` will be deprecated soon as it is ambiguous.